### DISCUSSION TITLE
Difference between F-Droid packages

### DISCUSSION THREAD

**Discussion by shuvashish76** [koreader/koreader#8651] [Q&A] [answered]
![Screenshot_2022-01-11-21-43-37](https://user-images.githubusercontent.com/48226312/148979611-9e0b842e-283c-4576-b7d1-9441d030737d.jpeg)
I see [two different apps](https://search.f-droid.org/?q=KOReader&lang=en) for KOReader on F-droid. 

1. `org.koreader.launcher.fdroid` - https://f-droid.org/en/packages/org.koreader.launcher.fdroid/
2. `org.koreader.launcher` - https://f-droid.org/en/packages/org.koreader.launcher/

Why two different packages & any major difference? (beside no1 is the latest one with additional `android.permission.MANAGE_EXTERNAL_STORAGE` & `android.permission.QUERY_ALL_PACKAGES` permissons)

**Comment by Frenzie** [comment]
See https://github.com/koreader/android-luajit-launcher/pull/318

As the version implies, the older one is older and it'll remain so. That's pretty much it.

@pazos How hard would it be to update the metadata for the older one to refer to the newer one?

**Reply by pazos** [reply, marked as answer]
It is on my/our todo list in https://github.com/koreader/koreader/issues/8225

I would like to remove the old package to not confuse users but I was told that's not desiderable. The best thing they can do is to archive the old package. It does involve one line change, as explained in https://gitlab.com/fdroid/fdroiddata/-/merge_requests/9318

**Comment by shuvashish76** [comment]
Please mention this information here: https://github.com/koreader/koreader/wiki/Installation-on-Android-devices

**Reply by pazos** [reply]
It is better to fix the underlying issue :)

**Comment by pazos** [comment]
> Why two different packages & any major difference? (beside no1 is the latest one with additional android.permission.MANAGE_EXTERNAL_STORAGE & android.permission.QUERY_ALL_PACKAGES permissons)

The new permissions are required since we bump targetSdk to API30.

MANAGE_EXTERNAL_STORAGE replaces READ/WRITE EXTERNAL STORAGE.
QUERY_ALL_PACKAGES is required to find 3rd party dict apps installed. Previous versions didn't require any runtime permission.
